### PR TITLE
refactor and reuse code for find by username and email

### PR DESCRIPTION
**Pull request by fredoche**
Hello, j'attend tes retours sur ce refacto effectué en binome.

### CODE REVIEW

File: app/models/User.scala [hunk 1]
```diff
@@ -7,24 +7,35 @@ import utils.Hash
...
+
+case class User(email: Option[User.Email],
+                passwordHash: Option[User.BcryptPassword],
                 id: String,
-                username: Option[String],
+                username: Option[User.UserName],
                 facebookToken: Option[String],
                 facebookUserId: Option[String],
                 creationDate: Date,
+                // public , private or unlisted. Specifies how the story can be retrieved
```
Comment (jdauphant) [inline app/models/User.scala]: My mistake : "public , private or unlisted" are for privacy in story and it's still in a PR.
This state on User is not used.

File: app/controllers/Users.scala [hunk 2]
```diff
@@ -121,24 +122,28 @@ object Users extends Controller with APIJsonFormats {
     }
   }
 
-  def checkEmail(email: String) = LoggingAction{
+  def checkEmail(email: User.Email) = LoggingAction{
       Action.async { request =>
         User.findByEmail(email).map {
-          case User(Some(`email`), _, _, _, _, _, _, _) :: Nil =>
-            Ok(Json.toJson(TopLevel(emails = Some(Left(Email(email, "registered"))))))
+            // match a list with a single user
```
Comment (jdauphant) [inline app/controllers/Users.scala]: This comment repeat the code.

File: app/controllers/Users.scala [hunk 3]
```diff
@@ -121,24 +122,28 @@ object Users extends Controller with APIJsonFormats {
...
+          case user :: Nil =>
+            Ok(Json.toJson(Map("emails" -> Email(email, "registered"))))
+            // match no result or a list bigger than one.
```
Comment (jdauphant) [inline app/controllers/Users.scala:130]: You can throw an exception like what you have done in "checkUsername".

File: app/models/User.scala (hunk 1 above)
Comment (fredoche) [inline app/models/User.scala]: on a qu'a fusionner ta PR avant.

Comment (jdauphant) [inline app/models/User.scala]: Yep you can remove the comment, it don't concern this class in any case.

**Pull request merged**